uk.co.compucorp.civicrm.hrcomments/CRM/HRComments/API/Query/CommentSelect.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+use Civi\API\SelectQuery;
+use CRM_HRComments_BAO_Comment as Comment;
+
+/**
+ * This class is basically a wrapper around Civi\API\SelectQuery.
+ */
+class CRM_HRComments_API_Query_CommentSelect {
+
+  /**
+   * @var array
+   *   An array of params passed to an API endpoint
+   */
+  private $params;
+
+  /**
+   * @var \Civi\API\SelectQuery
+   *  The SelectQuery instance wrapped by this class
+   */
+  private $query;
+
+  public function __construct($params) {
+    $this->params = $params;
+    $this->buildCustomQuery();
+  }
+
+  /**
+   * Build the custom query.
+   */
+  private function buildCustomQuery() {
+    $customQuery = CRM_Utils_SQL_Select::from(Comment::getTableName() . ' as a');
+
+    $this->addWhere($customQuery);
+    $this->filterReturnFields();
+    $this->query = new SelectQuery(Comment::class, $this->params, false);
+    $this->query->merge($customQuery);
+  }
+
+  /**
+   * Add the conditions to the query.
+   *
+   * This where it is ensured that only non soft-deleted comments are returned
+   * deleted contracts and with balance changes.
+   *
+   * @param \CRM_Utils_SQL_Select $customQuery
+   */
+  private function addWhere(CRM_Utils_SQL_Select $customQuery) {
+    $conditions[] = 'a.is_deleted = 0';
+    $customQuery->where($conditions);
+  }
+
+  /**
+   * Executes the query
+   *
+   * @return array
+   */
+  public function run() {
+    $results = $this->query->run();
+    return $results;
+  }
+
+  /**
+   * This function allows some fields to be filtered out of the query results.
+   * Currently only the is_deleted field is filtered out
+   */
+  private function filterReturnFields() {
+    if (empty($this->params['return'])) {
+      $this->params['return'] = ['id', 'entity_name', 'entity_id', 'text', 'contact_id', 'created_at'];
```

Comment:
what about getting the list of the filters with `Comment::fields()`, so you don't have to update this manually every time a new field gets added to the DAO?